1.29.0

matteobruni opened the pull request:
# 1.29.0

## New Features

- Added sizeOffset option to destroy split section
- Added inverse option to move gravity section
- Added fireworks preset package

## Bug Fixes

- Fixed move angle options
- Fixed move random and straight settings

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.3.1 to 8.3.2

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.1 to 8.3.2.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.1...8.3.2)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1616 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.3.2

build(deps-dev): bump postcss from 8.3.1 to 8.3.2

dependabot[bot] committed:
build(deps-dev): bump cssnano from 4.1.11 to 5.0.2

Bumps [cssnano](https://github.com/cssnano/cssnano) from 4.1.11 to 5.0.2.
- [Release notes](https://github.com/cssnano/cssnano/releases)
- [Commits](https://github.com/cssnano/cssnano/compare/v4.1.11...cssnano@5.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

code-factor committed:
[CodeFactor] Apply fixes

[ci skip] [skip ci]

matteobruni committed:
Merge pull request #1618 from matteobruni/dependabot/npm_and_yarn/dev/cssnano-5.0.2-cf-autofix

Apply fixes from CodeFactor

matteobruni committed:
Merge pull request #1518 from matteobruni/dependabot/npm_and_yarn/dev/cssnano-5.0.2

build(deps-dev): bump cssnano from 4.1.11 to 5.0.2

matteobruni committed:
build: fixed line shape rotation, it was 90 degrees out of phase

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
chore(release): published new version

 - ng-particles@2.11.1
 - inferno-particles@1.11.1
 - jquery-particles@1.28.1
 - preact-particles@1.28.1
 - react-tsparticles@1.28.1
 - riot-particles@1.2.1
 - solid-particles@1.11.1
 - svelte-particles@1.11.1
 - particles.vue@2.11.1
 - particles.vue3@1.11.1
 - web-particles@1.1.1
 - tsparticles-editor@1.28.1
 - tsparticles@1.28.1
 - ng-particles-demo@1.13.1
 - editor-demo@1.13.1
 - inferno-particles-demo@1.11.1
 - jquery-particles-demo@1.13.1
 - preact-particles-demo@1.13.1
 - react-tsparticles-demo@1.13.1
 - riot-particles-demo@1.2.1
 - solid-particles-demo@1.11.1
 - svelte-demo@1.11.1
 - tsparticles-demo@1.13.1
 - particles.vue-demo@1.13.1
 - particles.vue3-demo@1.11.1
 - web-particles-demo@1.13.1
 - tsparticles-preset-big-circles@1.13.1
 - tsparticles-preset-bubbles@1.13.1
 - tsparticles-preset-confetti@1.13.1
 - tsparticles-preset-fire@1.13.1
 - tsparticles-preset-firefly@1.13.1
 - tsparticles-preset-fountain@1.13.1
 - tsparticles-preset-links@1.13.1
 - tsparticles-preset-sea-anemone@1.13.1
 - tsparticles-preset-snow@1.13.1
 - tsparticles-preset-stars@1.13.1
 - tsparticles-shape-bubble@1.13.1
 - tsparticles-shape-confetti@1.13.1
 - tsparticles-shape-heart@1.13.1
 - tsparticles-shape-multiline-text@1.13.1
 - tsparticles-shape-rounded-rect@1.13.1
 - tsparticles-shape-spiral@1.13.1
 - cra-template-particles-typescript@1.11.1
 - cra-template-particles@1.11.1
 - tsparticles-website@1.28.1

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.0-beta.2 to 0.21.0-beta.3

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.0-beta.2 to 0.21.0-beta.3.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/commits)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.38.1 to 5.39.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.38.1 to 5.39.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.38.1...v5.39.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
build: prettier

dependabot[bot] committed:
build(deps): bump tslib from 2.2.0 to 2.3.0

Bumps [tslib](https://github.com/Microsoft/tslib) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/Microsoft/tslib/releases)
- [Commits](https://github.com/Microsoft/tslib/compare/2.2.0...2.3.0)

---
updated-dependencies:
- dependency-name: tslib
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1621 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.0-beta.3

build(deps-dev): bump typedoc from 0.21.0-beta.2 to 0.21.0-beta.3

matteobruni committed:
Merge pull request #1622 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.39.0

build(deps-dev): bump webpack from 5.38.1 to 5.39.0

matteobruni committed:
Merge pull request #1624 from matteobruni/dependabot/npm_and_yarn/dev/tslib-2.3.0

build(deps): bump tslib from 2.2.0 to 2.3.0

matteobruni committed:
build: website update

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
Merge pull request #1620 from matteobruni/dev

1.28.1

dependabot[bot] committed:
build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/angular

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/vue3

Bumps [postcss](https://github.com/postcss/postcss) from 7.0.35 to 7.0.36.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/7.0.35...7.0.36)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1625 from matteobruni/dependabot/npm_and_yarn/components/angular/postcss-7.0.36

build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/angular

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/vue3/postcss-7.0.36

matteobruni committed:
Merge pull request #1626 from matteobruni/dependabot/npm_and_yarn/components/vue3/postcss-7.0.36

build(deps): bump postcss from 7.0.35 to 7.0.36 in /components/vue3

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/jquery

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1627 from matteobruni/dependabot/npm_and_yarn/components/jquery/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/jquery

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/svelte

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/react

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1628 from matteobruni/dependabot/npm_and_yarn/components/svelte/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/svelte

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

dependabot[bot] committed:
build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/preact

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 5.1.1 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v5.1.1...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1630 from matteobruni/dependabot/npm_and_yarn/components/preact/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/preact

matteobruni committed:
Merge branch 'main' into dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

matteobruni committed:
Merge pull request #1629 from matteobruni/dependabot/npm_and_yarn/components/react/glob-parent-5.1.2

build(deps): bump glob-parent from 5.1.1 to 5.1.2 in /components/react

dependabot[bot] committed:
build(deps): bump gh-pages from 3.2.0 to 3.2.1

Bumps [gh-pages](https://github.com/tschaub/gh-pages) from 3.2.0 to 3.2.1.
- [Release notes](https://github.com/tschaub/gh-pages/releases)
- [Changelog](https://github.com/tschaub/gh-pages/blob/main/changelog.md)
- [Commits](https://github.com/tschaub/gh-pages/compare/v3.2.0...v3.2.1)

---
updated-dependencies:
- dependency-name: gh-pages
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump sass from 1.34.1 to 1.35.1

Bumps [sass](https://github.com/sass/dart-sass) from 1.34.1 to 1.35.1.
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.34.1...1.35.1)

---
updated-dependencies:
- dependency-name: sass
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/core from 7.14.5 to 7.14.6

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.14.5 to 7.14.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.14.6/packages/babel-core)

---
updated-dependencies:
- dependency-name: "@babel/core"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.3.2 to 8.3.4

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.2 to 8.3.4.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.2...8.3.4)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1631 from matteobruni/dependabot/npm_and_yarn/dev/gh-pages-3.2.1

build(deps): bump gh-pages from 3.2.0 to 3.2.1

matteobruni committed:
Merge pull request #1632 from matteobruni/dependabot/npm_and_yarn/dev/sass-1.35.1

build(deps): bump sass from 1.34.1 to 1.35.1

matteobruni committed:
Merge pull request #1633 from matteobruni/dependabot/npm_and_yarn/dev/babel/core-7.14.6

build(deps-dev): bump @babel/core from 7.14.5 to 7.14.6

matteobruni committed:
Merge pull request #1634 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.3.4

build(deps-dev): bump postcss from 8.3.2 to 8.3.4

matteobruni committed:
fix: fixed issue with themes/responsive options and canvas background

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: prettier

matteobruni committed:
Merge branch 'main' into dev

matteobruni committed:
fix: fixed themes

matteobruni committed:
chore(release): published new version

 - ng-particles@2.11.2
 - inferno-particles@1.11.2
 - jquery-particles@1.28.2
 - preact-particles@1.28.2
 - react-tsparticles@1.28.2
 - riot-particles@1.2.2
 - solid-particles@1.11.2
 - svelte-particles@1.11.2
 - particles.vue@2.11.2
 - particles.vue3@1.11.2
 - web-particles@1.1.2
 - tsparticles-editor@1.28.2
 - tsparticles@1.28.2
 - ng-particles-demo@1.13.2
 - editor-demo@1.13.2
 - inferno-particles-demo@1.11.2
 - jquery-particles-demo@1.13.2
 - preact-particles-demo@1.13.2
 - react-tsparticles-demo@1.13.2
 - riot-particles-demo@1.2.2
 - solid-particles-demo@1.11.2
 - svelte-demo@1.11.2
 - tsparticles-demo@1.13.2
 - particles.vue-demo@1.13.2
 - particles.vue3-demo@1.11.2
 - web-particles-demo@1.13.2
 - tsparticles-preset-big-circles@1.13.2
 - tsparticles-preset-bubbles@1.13.2
 - tsparticles-preset-confetti@1.13.2
 - tsparticles-preset-fire@1.13.2
 - tsparticles-preset-firefly@1.13.2
 - tsparticles-preset-fountain@1.13.2
 - tsparticles-preset-links@1.13.2
 - tsparticles-preset-sea-anemone@1.13.2
 - tsparticles-preset-snow@1.13.2
 - tsparticles-preset-stars@1.13.2
 - tsparticles-shape-bubble@1.13.2
 - tsparticles-shape-confetti@1.13.2
 - tsparticles-shape-heart@1.13.2
 - tsparticles-shape-multiline-text@1.13.2
 - tsparticles-shape-rounded-rect@1.13.2
 - tsparticles-shape-spiral@1.13.2
 - cra-template-particles-typescript@1.11.2
 - cra-template-particles@1.11.2
 - tsparticles-website@1.28.2

matteobruni committed:
build: website update

matteobruni committed:
Merge pull request #1637 from matteobruni/dev

1.28.0

dependabot[bot] committed:
build(deps): bump @angular/common from 12.0.4 to 12.0.5

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/common)

---
updated-dependencies:
- dependency-name: "@angular/common"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser-dynamic

Bumps [@angular/platform-browser-dynamic](https://github.com/angular/angular/tree/HEAD/packages/platform-browser-dynamic) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/platform-browser-dynamic)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser-dynamic"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump riot from 5.4.4 to 5.4.5

Bumps [riot](https://github.com/riot/riot) from 5.4.4 to 5.4.5.
- [Release notes](https://github.com/riot/riot/releases)
- [Commits](https://github.com/riot/riot/compare/v5.4.4...v5.4.5)

---
updated-dependencies:
- dependency-name: riot
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/parser from 4.26.1 to 4.27.0

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 4.26.1 to 4.27.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.27.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/animations from 12.0.4 to 12.0.5

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/animations)

---
updated-dependencies:
- dependency-name: "@angular/animations"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1638 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-12.0.5

build(deps): bump @angular/common from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1639 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-dynamic-12.0.5

build(deps): bump @angular/platform-browser-dynamic from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1640 from matteobruni/dependabot/npm_and_yarn/dev/riot-5.4.5

build(deps): bump riot from 5.4.4 to 5.4.5

matteobruni committed:
Merge pull request #1641 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/parser-4.27.0

build(deps-dev): bump @typescript-eslint/parser from 4.26.1 to 4.27.0

matteobruni committed:
Merge pull request #1642 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-12.0.5

build(deps): bump @angular/animations from 12.0.4 to 12.0.5

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-vue from 7.10.0 to 7.11.1

Bumps [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) from 7.10.0 to 7.11.1.
- [Release notes](https://github.com/vuejs/eslint-plugin-vue/releases)
- [Commits](https://github.com/vuejs/eslint-plugin-vue/compare/v7.10.0...v7.11.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-vue
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/platform-browser from 12.0.4 to 12.0.5

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/platform-browser)

---
updated-dependencies:
- dependency-name: "@angular/platform-browser"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.51.1 to 2.52.0

Bumps [rollup](https://github.com/rollup/rollup) from 2.51.1 to 2.52.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.51.1...v2.52.0)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.3.4 to 8.3.5

Bumps [postcss](https://github.com/postcss/postcss) from 8.3.4 to 8.3.5.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.3.4...8.3.5)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/forms from 12.0.4 to 12.0.5

Bumps [@angular/forms](https://github.com/angular/angular/tree/HEAD/packages/forms) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/forms)

---
updated-dependencies:
- dependency-name: "@angular/forms"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1643 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-vue-7.11.1

build(deps-dev): bump eslint-plugin-vue from 7.10.0 to 7.11.1

matteobruni committed:
Merge pull request #1644 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-12.0.5

build(deps): bump @angular/platform-browser from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1645 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.0

build(deps-dev): bump rollup from 2.51.1 to 2.52.0

matteobruni committed:
Merge pull request #1646 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.3.5

build(deps-dev): bump postcss from 8.3.4 to 8.3.5

dependabot[bot] committed:
build(deps): bump @angular/core from 12.0.4 to 12.0.5

Bumps [@angular/core](https://github.com/angular/angular/tree/HEAD/packages/core) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/core)

---
updated-dependencies:
- dependency-name: "@angular/core"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typedoc from 0.21.0-beta.3 to 0.21.0-beta.4

Bumps [typedoc](https://github.com/TypeStrong/TypeDoc) from 0.21.0-beta.3 to 0.21.0-beta.4.
- [Release notes](https://github.com/TypeStrong/TypeDoc/releases)
- [Commits](https://github.com/TypeStrong/TypeDoc/commits)

---
updated-dependencies:
- dependency-name: typedoc
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump sass-loader from 12.0.0 to 12.1.0

Bumps [sass-loader](https://github.com/webpack-contrib/sass-loader) from 12.0.0 to 12.1.0.
- [Release notes](https://github.com/webpack-contrib/sass-loader/releases)
- [Changelog](https://github.com/webpack-contrib/sass-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/sass-loader/compare/v12.0.0...v12.1.0)

---
updated-dependencies:
- dependency-name: sass-loader
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1647 from matteobruni/dependabot/npm_and_yarn/dev/angular/forms-12.0.5

build(deps): bump @angular/forms from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1648 from matteobruni/dependabot/npm_and_yarn/dev/angular/core-12.0.5

build(deps): bump @angular/core from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1649 from matteobruni/dependabot/npm_and_yarn/dev/typedoc-0.21.0-beta.4

build(deps-dev): bump typedoc from 0.21.0-beta.3 to 0.21.0-beta.4

matteobruni committed:
Merge pull request #1651 from matteobruni/dependabot/npm_and_yarn/dev/sass-loader-12.1.0

build(deps-dev): bump sass-loader from 12.0.0 to 12.1.0

dependabot[bot] committed:
build(deps-dev): bump @angular/compiler-cli from 12.0.4 to 12.0.5

Bumps [@angular/compiler-cli](https://github.com/angular/angular/tree/HEAD/packages/compiler-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/compiler-cli)

---
updated-dependencies:
- dependency-name: "@angular/compiler-cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.26.1 to 4.27.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.27.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/router from 12.0.4 to 12.0.5

Bumps [@angular/router](https://github.com/angular/angular/tree/HEAD/packages/router) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/router)

---
updated-dependencies:
- dependency-name: "@angular/router"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1650 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.27.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.26.1 to 4.27.0

matteobruni committed:
Merge pull request #1653 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-cli-12.0.5

build(deps-dev): bump @angular/compiler-cli from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1655 from matteobruni/dependabot/npm_and_yarn/dev/angular/router-12.0.5

build(deps): bump @angular/router from 12.0.4 to 12.0.5

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.52.0 to 2.52.1

Bumps [rollup](https://github.com/rollup/rollup) from 2.52.0 to 2.52.1.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.52.0...v2.52.1)

---
updated-dependencies:
- dependency-name: rollup
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1654 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.52.1

build(deps-dev): bump rollup from 2.52.0 to 2.52.1

matteobruni committed:
feat: added sizeOffset option to destroy split section
feat: added inverse option to move gravity section
fix: fixed move angle options
fix: fixed move random and straight settings
build: added fireworks demo preset

matteobruni committed:
feat: added fireworks preset

ghost commented:
## DeepCode's analysis on [#61a9fc](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8d6b625346cf7950e2ea0b0fa51321d2cc7e5507/matteobruni/tsparticles/61a9fcf0dafd6f976ee805a7f346e248cd31b466/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) found:

- :warning: **1** warning :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/61a9fcf0dafd6f976ee805a7f346e248cd31b466/demo/vanilla/app.js#L42">app.js:42</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/61a9fcf0dafd6f976ee805a7f346e248cd31b466/demo/vanilla/app.js#L46">app.js:46</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/61a9fcf0dafd6f976ee805a7f346e248cd31b466/demo/vanilla/app.js#L50">app.js:50</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8d6b625346cf7950e2ea0b0fa51321d2cc7e5507/matteobruni/tsparticles/61a9fcf0dafd6f976ee805a7f346e248cd31b466/pr/_/%2Fdemo%2Fvanilla%2Fapp.js/javascript%2Fdc_interfile_project%2FNoRateLimitingForExpensiveWebOperation/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/8d6b625346cf7950e2ea0b0fa51321d2cc7e5507/matteobruni/tsparticles/61a9fcf0dafd6f976ee805a7f346e248cd31b466/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)
#### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 12.0.4 to 12.0.5

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.4...v12.0.5)

---
updated-dependencies:
- dependency-name: "@angular/cli"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/compiler from 12.0.4 to 12.0.5

Bumps [@angular/compiler](https://github.com/angular/angular/tree/HEAD/packages/compiler) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/12.0.5/packages/compiler)

---
updated-dependencies:
- dependency-name: "@angular/compiler"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.39.0 to 5.39.1

Bumps [webpack](https://github.com/webpack/webpack) from 5.39.0 to 5.39.1.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.39.0...v5.39.1)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1658 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-12.0.5

build(deps-dev): bump @angular/cli from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1660 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-12.0.5

build(deps): bump @angular/compiler from 12.0.4 to 12.0.5

matteobruni committed:
Merge pull request #1662 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.39.1

build(deps-dev): bump webpack from 5.39.0 to 5.39.1

dependabot[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 12.0.4 to 12.0.5.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v12.0.4...v12.0.5)

---
updated-dependencies:
- dependency-name: "@angular-devkit/build-angular"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1661 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-12.0.5

build(deps-dev): bump @angular-devkit/build-angular from 12.0.4 to 12.0.5

matteobruni committed:
feat: emitter rate delay and quantity are now RangeValues so they can be randomized

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

matteobruni committed:
build: updated deps
